### PR TITLE
Access request locks

### PR DESCRIPTION
**Pull request by espadolini**
This adds a way to add a lock on a specific access request; when such a lock is in force, any identity that has used that access request is prevented from interacting with the system - the user can however switch to a less privileged certificate, or log out and log back in again and regain their own regular access.

While it would be sufficient to lock the user out if malicious behavior was suspected, this helps in the case where elevated privileges are requested and granted by mistake; the request can be blocked and the user can then continue working with their own static roles (or with the roles granted by other access requests).

The `types.LockTarget` type will gain a new `AccessRequest` field, following the nomenclature of `AccessRequests`/`access_requests` already in use in the `session.start` event. 

As locks are currently only supported on the tool side by `tctl`, the only addition to the UX is a new flag to `tctl lock`
```
tctl lock --access-request=261e80c5-357b-4c43-9b67-40a6bc4c6e4d
```
that matches the field name in `types.LockTarget` (converted to `kebab-case` as the other CLI options are).

Such a lock serialized as yaml (as output by `tctl get locks` or `tctl get lock/<lockname>`) looks like
```
kind: lock
metadata:
  id: 1640275041241490000
  name: 7f6d2edc-68cb-4b98-b2d4-859c75284c8a
spec:
  target:
    access_request: 261e80c5-357b-4c43-9b67-40a6bc4c6e4d
version: v2
```

Fixes TEL-Q321-3.

Docs PR:
* https://github.com/gravitational/teleport/pull/9866

### CODE REVIEW

File: lib/auth/auth.go [hunk 1]
```diff
@@ -1740,6 +1749,7 @@ func (a *Server) ExtendWebSession(req WebSessionReq, identity tlsca.Identity) (t
 
 		roles = append(roles, newRoles...)
 		roles = apiutils.Deduplicate(roles)
+		accessRequests = apiutils.Deduplicate(append(accessRequests, req.AccessRequestID))
```
Comment (fspmarshall) [inline lib/auth/auth.go:1752]: `AccessRequestID` may be `""`.  The append should be conditional.

Comment (espadolini) [reply lib/auth/auth.go:1752]: We're in `if req.AccessRequestID != "" {` here

**Comment by russjones** [comment]
@espadolini Can you update the PR description with details on CLI UX?

**Comment by espadolini** [comment]
The underlying mechanism for rolling back an approved access request must necessarily be a lock, as the certificates issued  while the request was approved would otherwise stay valid even if the access request itself is invalidated. It's already possible to delete a request or to forcibly set a request with an `APPROVED` status to `DENIED` with edit/delete privileges.

Once we're able to issue locks on access requests, we can decide if a lock should be created automatically when an approved and not expired request becomes denied or gets deleted.

There's no RFD for this, the scope didn't seem big enough to justify one.

**Comment by codingllama** [comment]
> The underlying mechanism for rolling back an approved access request must necessarily be a lock, as the certificates issued while the request was approved would otherwise stay valid even if the access request itself is invalidated. It's already possible to delete a request or to forcibly set a request with an APPROVED status to DENIED with edit/delete privileges.

Makes sense, thanks for the explanation. I still find it strange to lock an approval request, but I have no qualms with _some_ lock existing.

> Once we're able to issue locks on access requests, we can decide if a lock should be created automatically when an approved and not expired request becomes denied or gets deleted.

I'm very much of this opinion, otherwise we are relying on a somewhat deep understanding of Teleport from admins.

> There's no RFD for this, the scope didn't seem big enough to justify one.

Ack.

**Review by codingllama — APPROVED** [review]

**Review by fspmarshall — APPROVED** [review]

**Review by lxea — APPROVED** [review]

**Pull request merged**